### ISSUE TITLE
Cheat Engine doesn't search whole process

### ISSUE THREAD

**Issue by the-pirate-bay** [cheat-engine/cheat-engine#239]
First of all, thank you very much for writing open source software. Cheat Engine is a great project.

Here is the actual issue I found. It is possible that maybe I didn't understand properly how Cheat Engine works, but it seems that Cheat Engine doesn't search whole process for strings. Some memory regions are simply skipped without any warnings.

The easiest way to reproduce it is to follow these steps:
1. Launch Cheat Engine (64 bit)
2. Set `Scan Value` to `String` and `Memory Scan Options -> Writable` to `Gray`
3. Open Google Chrome (64 bit) and set Chrome's language to English
4. Open each Chrome's process (each `chrome.exe`) and search for the string `to exit full screen`
5. Then for each Chrome's process search same text, but unicode (add null-byte after each char, and switch to `array of bytes`)
6. You will find nothing in both (4) and (5), except if you recently visited a page containing string `"to exit full screen"`. If you find anything in (4) or (5), try clearing the cache, restarting Chrome and try again
7. Now take a mamory dump of each Chrome's process and inspect each file using some hex editor and search for the same string inside dump file. You will find a few occurences.
8. Now, find the offset in the dump file, go back to Cheat engine and `Browse this memory region` for that offset. You will find string `"to exit full screen"` in Chrome's memory.

So, basically, in steps (4) and (5), searching for string `"to exit full screen"` doesn't display any results, but in the steap (8) you can clearly see that such a string is indeed located in Chrome's memory.

If you cannot reproduce it for `chrome.exe` and string `"to exit full screen"`, then try it for `explorer.exe` (windows eplorer) and string `"if you change a file name extension"`. Take the same steps, you will find the string inside explorer's memory, but Cheat Engine is not finding it.

**Comment by cheat-engine** [comment]
is copy on write also grey ?
is mem_mapped memory checkked in settings->scan settings ?

**Comment by the-pirate-bay** [comment]
Checking the option `mem_mapped` actually solved the problem, because these regions are files mapped by the system. Didn't know about it. Thanks!
